G1-true-binary-metadata.md

```diff
@@ -0,0 +1,89 @@
+True Binary Metadata
+----
+* Author(s): ctiller
+* Approver: ejona
+* Status: Draft
+* Implemented in: n/a
+* Last updated: 3/29/17
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/Ww8waYz_Nes
+
+## Abstract
+
+Add a HTTP2 extension to allow binary metadata (those suffixed with -bin) to be
+sent without the base64 encode/decode step.
+
+## Background
+
+gRPC allows binary metadata to be sent by applications. These metadata elements
+are keyed with a -bin suffix. When transmitted on the wire, since HTTP2 does not
+allow binary headers, we base64 encode these elements and Huffman compress them.
+On receipt, we transparently reverse the transformation.
+
+This transformation is costly in terms of CPU, and there exist use cases where
+this transformation can become the CPU bottleneck for gRPC.
+
+### Related Proposals:
+
+n/a
+
+## Proposal
+
+### New setting
+
+Expose a custom setting in our HTTP2 settings exchange:
+GRPC_ALLOW_TRUE_BINARY_METADATA = 0xfe03.
```

Comment:
Why only `-bin` headers? If we're using `0x00` magic prefix, then I don't see a reason for this limitation.